### PR TITLE
copyto! for broadcasting

### LINKED ISSUE #71: Broadcasting with .= does not error when names don't match
I think you know this, but since I can't find an issue, note that in-place broadcasting seems to ignore names:
```julia
ab = NamedDimsArray(rand(2,2), (:a, :b));
ba = NamedDimsArray(rand(2,2), (:b, :a));
ac = NamedDimsArray(rand(2,2), (:a, :c));
z = zeros(2,2);

ab .+ ba'         # ok!

ac .= ab .+ ba    # no error, not ok

z .= ab .+ ac     # also no error
```

### PR DESCRIPTION
**Pull request by mcabbott**
Closes #71

### CODE REVIEW

File: src/broadcasting.jl [hunk 1]
```diff
@@ -55,7 +55,13 @@ function Broadcast.copy(bc::Broadcasted{NamedDimsStyle{S}}) where S
     L = broadcasted_names(bc)
     return NamedDimsArray{L}(data)
 end
-# TODO: copyto! for broadcasting
```
Comment (oxinabox) [inline src/broadcasting.jl]: I blame past me for this.

File: test/broadcasting.jl [hunk 2]
```diff
@@ -89,6 +93,24 @@ using Tracker
         @test names(nda .+ ones(1,20)) == (:x, :y, :z)
     end
 
+    @testset "in-place" begin
```
Comment (oxinabox) [inline test/broadcasting.jl]:
```suggestion
    @testset "in-place assignment .=" begin
```

File: test/broadcasting.jl [hunk 3]
```diff
@@ -89,6 +93,24 @@ using Tracker
...
+        ab = NamedDimsArray(rand(2,2), (:a, :b));
+        ba = NamedDimsArray(rand(2,2), (:b, :a));
+        ac = NamedDimsArray(rand(2,2), (:a, :c));
+        z = zeros(2,2);
+        @test names(ab .+ ba' .+ z) == (:a, :b)
+        @test_throws DimensionMismatch ab .+ ba
```
Comment (oxinabox) [inline test/broadcasting.jl]: Don't we have tests for this above?

File: test/broadcasting.jl [hunk 4]
```diff
@@ -89,6 +93,24 @@ using Tracker
         @test names(nda .+ ones(1,20)) == (:x, :y, :z)
     end
 
+    @testset "in-place" begin
+        ab = NamedDimsArray(rand(2,2), (:a, :b));
+        ba = NamedDimsArray(rand(2,2), (:b, :a));
+        ac = NamedDimsArray(rand(2,2), (:a, :c));
+        z = zeros(2,2);
```
Comment (oxinabox) [inline test/broadcasting.jl]: what is with all the extra semicolons?

File: test/broadcasting.jl [hunk 5]
```diff
@@ -89,6 +93,24 @@ using Tracker
...
+        @test names(ab .+ ba' .+ z) == (:a, :b)
+        @test_throws DimensionMismatch ab .+ ba
+
+        # https://github.com/invenia/NamedDims.jl/issues/71
+        @test_throws DimensionMismatch z .= ab .+ ba
+        @test_throws DimensionMismatch z .= ab .+ ac
+        @test_throws DimensionMismatch ac .= ab .+ ba
+
+        # check that dest is written into:
+        @test names(z .= ab .+ ba') == (:a, :b)
```
Comment (oxinabox) [inline test/broadcasting.jl]:
```suggestion
        @test names(z .= ab .+ ba') == (:a, :b)
        @test z isa Array  # has not itself magically gained names
```

File: test/broadcasting.jl (hunk 4 above)
Comment (mcabbott) [reply test/broadcasting.jl]: I don't know, but gone now!

File: src/broadcasting.jl (hunk 1 above)
Comment (mcabbott) [reply src/broadcasting.jl]: A likely suspect.

**Pull request merged**